Infrastructure

Lithoid opened the pull request.

Commit:
Added repositories

Commit:
Created references in context

Commit:
Added entitites

dmmitrenko commented on Evernote.Repositories/Abstract/DbRepository.cs:
```diff
@@ -0,0 +1,99 @@
+
+        public async Task<bool> AddItemAsync(T entity, bool saving = true) {
+            _context.Set<T>().Add(entity);
+            if (saving)
```
А для чого потрібна змінна saving?

dmmitrenko commented on Evernote.Repositories/Abstract/DbRepository.cs:
```diff
@@ -0,0 +1,99 @@
+{
+    public class DbRepository<T> : IDbRepository<T>
+        where T : class, IDbEntity {
+        private DbContext _context;
```
Ще потрібно додати пропертю _dbSet<T>, а не кожного разу робити в методах _context.Set<T>();

dmmitrenko replied:
Ще потрібно додати поле _dbSet, а не кожного разу робити в методах _context.Set();

dmmitrenko commented on Evernote.Repositories/Abstract/DbRepository.cs:
```diff
@@ -0,0 +1,99 @@
+            return await SaveChangesAsync() > 0;
+        }
+
+        public async Task<T> GetItemAsync(Expression<Func<T, bool>>? filter = null, bool tracked = true, string includeProps = null) {
```
Як на мене, складна логіка для простого метода репозиторія.
По-перше filter - обов'язковий параметр, навіщо ставити йому за замовчуванням null.
Не розумію навіщо Where, коли можна одразу FirstOrDefault(filter), тоді й перевірки на null не потрібно.
На рахунок інклюдів, можна в параметрах приймати який саме інклюд потрібен (або якщо декілька, можна написати свій екстенш метод).
Також ніколи не чув за ситуації , коли нам потрібен був tracking для того, щоб просто отримати entity. Думаю, потрібно вже за замовчуванням додавати до запиту AsNoTracking()

dmmitrenko replied:
Як на мене, складна логіка для простого метода репозиторія.
По-перше filter - обов'язковий параметр, навіщо ставити йому за замовчуванням null.
Не розумію навіщо Where, коли можна одразу FirstOrDefault(filter), тоді й перевірки на null не потрібно.
На рахунок інклюдів, можна в параметрах приймати який саме інклюд потрібен (або якщо декілька, можна написати свій екстенш метод).
Також ніколи не чув за ситуації , коли нам потрібен був tracking для того, щоб просто отримати entity. Думаю, потрібно вже за замовчуванням додавати до запиту AsNoTracking()

dmmitrenko commented on Evernote.Repositories/Abstract/DbRepository.cs:
```diff
@@ -0,0 +1,99 @@
+            }
+        }
+
+        public async Task<List<T>> GetAllItemsAsync(Expression<Func<T, bool>> filter = null, string includeProps = null) {
```
Аналогічно і тут для інклюдів, я би краще приймав ті інклюди які необхідні у вигляді params. Логіка, яка тут інтуїтивно не зрозуміла.

BaseCampGL2023 commented on Evernote.Context/AppDataContext.cs:
```diff
@@ -1,11 +1,52 @@
+                    note => {
+
+
+                        note.HasOne(n => n.User)
```
I would suggest using ApplyConfiguration and Configs in a separate file.
This is just styling stuff - no need to change

BaseCampGL2023 commented on Evernote.Repositories/Abstract/DbRepository.cs:
```diff
@@ -0,0 +1,99 @@
+
+
+
+            try {
```
A little bit strange construction. Better to leave as is without try/catch. Later on in code we may want to react on exception and in this case we would need to always check the output for '-1'

Commit:
Fixes after review

Commit:
- Changed the architecture of the project (some layers have been merged, and some have been renamed to more accurate names)
- Slightly changed the logic of some methods of the base repository
- New business logic for notes
- Added NoteController

lannahirave committed:
Merge pull request #3 from GLBaseCamp2023/feature/note-api

Feature/note api
